### PR TITLE
fix: add curl module

### PR DESCRIPTION
**Pull request by fengshunli**
Signed-off-by: fengshunli <1171313930@qq.com>

### CODE REVIEW

**Review by tao12345666333 — COMMENTED** [review]
If we add curl, we can directly replace `wget` with` curl`, remove `wget`, WDYT?

**Comment by tokers** [comment]
> If we add curl, we can directly replace `wget` with` curl`, remove `wget`, WDYT?

Someone, or some programs might rely on `wget`, I think keep both of them will be better.

**Comment by tao12345666333** [comment]
> Someone, or some programs might rely on wget, I think keep both of them will be better.

In the APISIX scenario, since the admin API call needs to be triggered, it is normal to add curl, and curl is also used in our documents.

But for `wget`, there is no use case other than downloading files in the APISIX scenario, can you provide some scenarios? Otherwise, I'm still keeping my opinion.

Minimize unnecessary package installation in the image to reduce the threat of vulnerabilities and reduce the size of the image.

File: debian/Dockerfile [hunk 1]
```diff
@@ -22,18 +22,18 @@ ARG APISIX_VERSION=3.1.0
...
       amd64) \
         echo "deb http://openresty.org/package/debian $codename openresty" | tee /etc/apt/sources.list.d/openresty.list \
-        && wget -O - http://repos.apiseven.com/pubkey.gpg | apt-key add - \
+        && curl -sSL http://repos.apiseven.com/pubkey.gpg | sudo apt-key add - \
         && echo "deb http://repos.apiseven.com/packages/debian $codename main" | tee /etc/apt/sources.list.d/apisix.list \
         ;; \
       arm64) \
         echo "deb http://openresty.org/package/arm64/debian $codename openresty" | tee /etc/apt/sources.list.d/openresty.list \
-        && wget -O - http://repos.apiseven.com/pubkey.gpg | apt-key add - \
+        && curl -sSL http://repos.apiseven.com/pubkey.gpg | sudo apt-key add - \
```
Comment (tao12345666333) [inline debian/Dockerfile:36]: This modification is wrong.   `sudo` is not needed here, and `sudo` doesn't exist.

Please build successfully locally before submitting, thank you

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: debian/Dockerfile (hunk 1 above)
Comment (fengshunli) [reply debian/Dockerfile]: updated
image

**Comment by tokers** [comment]
> > Someone, or some programs might rely on wget, I think keep both of them will be better.
> 
> In the APISIX scenario, since the admin API call needs to be triggered, it is normal to add curl, and curl is also used in our documents.
> 
> But for `wget`, there is no use case other than downloading files in the APISIX scenario, can you provide some scenarios? Otherwise, I'm still keeping my opinion.
> 
> Minimize unnecessary package installation in the image to reduce the threat of vulnerabilities and reduce the size of the image.

I have no specific scenario but when we debug issues in the container, we (or our tools) may rely on the wget or curl.

File: debian/Dockerfile [hunk 2]
```diff
@@ -22,18 +22,18 @@ ARG APISIX_VERSION=3.1.0
 RUN set -ex; \
     arch=$(dpkg --print-architecture); \
     apt update; \
-    apt-get -y install --no-install-recommends wget gnupg ca-certificates; \
+    apt-get -y install --no-install-recommends wget gnupg ca-certificates curl; \
```
[the remark this answers is not in the document]
Comment (fengshunli) [reply debian/Dockerfile:25]: Are you sure you want to delete this wget? @Baoyuantop

**Comment by tao12345666333** [comment]
> I have no specific scenario but when we debug issues in the container, we (or our tools) may rely on the wget or curl.

I have yet to see anyone using `wge`t instead of `curl` in APISIX's debugging scenario.

`curl `will be installed in this PR. It is enough for users' debugging and basic use.

File: debian/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (Baoyuantop) [reply debian/Dockerfile]: I think we can remove it. cc @tao12345666333

**Review by tokers — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**